4 build set up deployment

gabrielcnunez opened the pull request:
<!-- Erase any parts of this template not applicable to your Pull Request unless marked required. -->
## What Type of PR is this? 
- [ ] 🎁 Feature
- [ ] 🐞 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🛠️ Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [X] 🤖 Build / CI/ DevOps
- [ ] ⏪️ Revert 
- [ ] 🙋 Request for Help
- [ ] 💎 Adds dependencies
- [ ] 📊 Changes DB
- [ ] 👽 Other: _explain here_
## Describe your changes <!-- Required -->
  <sup>Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.</sup>

App was deployed using Fly.io and a root view was added to `config/routes.rb` to confirm successful deployment (this can be removed once API development begins).  In addition, the `rubyonrails.yml` was updated to add continuous deployment to Fly.io when changes are pushed onto the main branch.

## Issue number and link <!-- Required -->
<sup>Only PR's for existing issue tickets or features for this sprint will be approved. Issue _MUST_ be linked for review.</sup>
<!-- Put `closes #XXXX` or 'fixes #XXXX' in your comment to auto-close the PR issue on merge-->
Closes #4 issue
## Questions for reviewers?
<sup>Please note anything you would like specific feedback on in your code.</sup>


## Checklist before requesting a review <!-- Required -->

| Repo Contribution Standards| Complete |
| ----- | ---------------- |
| I have performed a self-review of my code | <ul><li> [X] **Yes** </li><li> [ ] **No**</li> |
| My code follows the style guidelines of this project| <ul><li> [X] **Yes** </li><li> [ ] **No**</li> |
| I have made corresponding changes to the documentation (Note: Documentation _must_ be up to date for merge)| <ul><li> [X] **Yes** </li><li> [ ] **No**</li> |

| Change Type | Complete |
| ----- | ---------------- |
| Non-breaking change which fixes an issue | <ul><li> [X] **Yes** </li><li> [ ] **No**</li> |
| Breaking change (fix or feature that would cause existing functionality to not work as expected) | <ul><li> [ ] **Yes** </li><li> [X] **No**</li> |

| Testing | Complete |
| ----- | ---------------- |
| I have added tests that prove my fix is effective or that my feature works | <ul><li> [ ] **Yes** </li><li> [X] **No**</li>* |
| New and existing tests pass locally with my changes | <ul><li> [X] **Yes** </li><li> [ ] **No**</li> |

## Any additional collaborators for this PR?
<!-- Tag anyone else on the project that can answer questions on this PR -->
No
## Any Specific Reviewer approval needed?
<!-- Tag anyone who must give approval on PR prior to merge -->
@aleish-m

aleish-m commented:
Hey, @gabrielcnunez I see the linting Failed with Rubocop. 
🫣 Can you make the following changes on your branch? I want to make sure that it is up and running smoothly on more than just my end, or I'd make them and push the changes.

1. In the `rubocop.yml` file:
     Add `NewCops: enable` under `line 7` "AllCops:"
2. Run `rubocop` in your terminal, and this should give all the errors that RuboCop will find when pushed. It looks like the current issues are all double rather than single quote errors, which can be auto-corrected. 
   - If you want to Auto Correct all possible errors in the whole repo, run `robocop -a` in your terminal anything not auto-corrected will still be listed for you to review and fix manually. (ref: https://docs.rubocop.org/rubocop/1.54/usage/basic_usage.html ) 
   - If you are using VSCode and want to Auto-correct 1 file at a time. You can open that file and hit `cmd+opt+B`, and that should run auto-formatting on that one file.
3. In the pull request template, can you please change the `| **Yes** [ ] <br> **No** [ ] |` in the checklists to `| <ul> <li> [ ] **Yes** </li> <li> [ ] **No** </li> |` that way the checkboxes format correctly.
4. Add the deployed site link to the ReadMe doc, so we all have that link easy to access and visit.
    - Also, here is the MD Code for the Deployment status badge from GitHub Actions: So that specific aspect can be shown separately from CI if you want. 🙂
```
[![Ruby on Rails CD](https://github.com/Plan-My-Dinner/plan_my_dinner_be/actions/workflows/rubyonrails.yml/badge.svg?event=deployment_status)](https://github.com/Plan-My-Dinner/plan_my_dinner_be/actions/workflows/rubyonrails.yml)
```

✅ Otherwise, all the deployment files look great, and I know you showed it up and running. 🔥🔥🔥🔥🔥🔥🔥

gabrielcnunez commented:
> Hey, @gabrielcnunez I see the linting Failed with Rubocop. 🫣 Can you make the following changes on your branch? I want to make sure that it is up and running smoothly on more than just my end, or I'd make them and push the changes.
> 
> 1. In the `rubocop.yml` file:
>    Add `NewCops: enable` under `line 7` "AllCops:"
> 2. Run `rubocop` in your terminal, and this should give all the errors that RuboCop will find when pushed. It looks like the current issues are all double rather than single quote errors, which can be auto-corrected.
>    
>    * If you want to Auto Correct all possible errors in the whole repo, run `robocop -a` in your terminal anything not auto-corrected will still be listed for you to review and fix manually. (ref: https://docs.rubocop.org/rubocop/1.54/usage/basic_usage.html )
>    * If you are using VSCode and want to Auto-correct 1 file at a time. You can open that file and hit `cmd+opt+B`, and that should run auto-formatting on that one file.
> 3. In the pull request template, can you please change the `| **Yes** [ ] <br> **No** [ ] |` in the checklists to `| <ul> <li> [ ] **Yes** </li> <li> [ ] **No** </li> |` that way the checkboxes format correctly.
> 4. Add the deployed site link to the ReadMe doc, so we all have that link easy to access and visit.
>    
>    * Also, here is the MD Code for the Deployment status badge from GitHub Actions: So that specific aspect can be shown separately from CI if you want. 🙂
> 
> ```
> [![Ruby on Rails CD](https://github.com/Plan-My-Dinner/plan_my_dinner_be/actions/workflows/rubyonrails.yml/badge.svg?event=deployment_status)](https://github.com/Plan-My-Dinner/plan_my_dinner_be/actions/workflows/rubyonrails.yml)
> ```
> 
> ✅ Otherwise, all the deployment files look great, and I know you showed it up and running. 🔥🔥🔥🔥🔥🔥🔥

@aleish-m Thank you for doing a deep dive on this review!  I'll address your items in order:

1. `NewCops: enable` was added to the `rubocop.yml` file
2. Ran rubocop in the terminal and had it autocorrect the issues (including changing double quotes to single like you mentioned)
3. Updated the pull request template with the formatting as you described
4. Added the link to the deployed website, but we tweak things further once we start building out the README in earnest

Let me know if everything looks good, thanks!

aleish-m reviewed (approved):
Thanks for making those changes. All looks good to me at this point. Yes, the ReadME needs to be worked on overall, but at least with the link to the deployed site, we all have access to the URL as needed.
